app/controllers/Checkout.scala

```diff
@@ -216,26 +205,26 @@ object Checkout extends Controller with LazyLogging with ActivityTracking with C
     }
   }
 
-  def thankYouDigipack = NoCacheAction.async { request => thankYou(Digipack).apply(request) }
-
-  def thankYouPaper = AuthorisedTester.async { request => thankYou(Paper).apply(request) }
-
-  def thankYou(product: ProductFamily) = NoCacheAction { implicit request =>
 
+  def thankYou() = NoCacheAction { implicit request =>
     implicit val resolution: TouchpointBackend.Resolution = TouchpointBackend.forRequest(PreSigninTestCookie, request.cookies)
     implicit val tpBackend = resolution.backend
+
+    import tpBackend.catalogService._
+
     val session = request.session
 
     val sessionInfo = for {
       subsName <- session.get(SubsName)
+      plan <- session.get(RatePlanId).map(ProductRatePlanId).flatMap(p => paperCatalog.flatMap(_.find(p)).orElse(digipackCatalog.find(p)))
       ratePlanId <- session.get(RatePlanId)
       currencyStr <- session.get(SessionKeys.Currency)
       currency <- Currency.fromString(currencyStr)
-    } yield (subsName, ratePlanId, currency)
+    } yield (subsName, plan, currency)
 
     sessionInfo.fold {
-      Redirect(getCheckoutRouteForProductFamily(product = product)).withNewSession
-    } { case (subsName, ratePlanId, currency) =>
+      Redirect(routes.Homepage.index()).withNewSession
```

Comment:
Hopefully this is okay - we just go to the homepage if you visit the thank you page with no plan to simplify the logic here, because product family doesn't differentiate between paper & digipack